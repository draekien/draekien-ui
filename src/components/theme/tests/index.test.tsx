/** @jsx jsx */
import { jsx } from 'theme-ui';
import { render } from '@testing-library/react';
import { ThemeProvider } from '..';

describe('Theme component', () => {
  test('should merge themes but not dangerous keys', () => {
    const component = render(
      <ThemeProvider
        theme={{
          space: {
            xl: '5rem',
          },
          // @ts-ignore
          __proto__: 'something bad',
          // @ts-ignore
          constructor: 'something worse',
        }}
      >
        <div />
      </ThemeProvider>
    );

    expect(component.container).toMatchSnapshot();
  });
});
