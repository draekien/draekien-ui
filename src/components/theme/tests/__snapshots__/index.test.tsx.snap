// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme component should merge themes but not dangerous keys 1`] = `
<div>
  <div />
</div>
`;
